src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/TenantEmailTemplateService.java

```diff
@@ -32,30 +35,53 @@ public class TenantEmailTemplateService implements RefreshableConfiguration {
     private final String pathPattern;
     private final String customEmailPathPattern;
     private final StringTemplateLoader templateLoader;
+    private final MultiTenantLangStringTemplateLoaderService multiTenantLangStringTemplateLoaderService;
+    private final EmailSpecService emailSpecService;
 
     public TenantEmailTemplateService(ApplicationProperties applicationProperties,
-                                      StringTemplateLoader templateLoader) {
+                                      StringTemplateLoader templateLoader,
+                                      MultiTenantLangStringTemplateLoaderService multiTenantLangStringTemplateLoaderService,
+                                      EmailSpecService emailSpecService) {
         this.templateLoader = templateLoader;
+        this.multiTenantLangStringTemplateLoaderService = multiTenantLangStringTemplateLoaderService;
         this.pathPattern = applicationProperties.getEmailPathPattern();
         this.customEmailPathPattern = applicationProperties.getCustomEmailPathPattern();
+        this.emailSpecService = emailSpecService;
     }
 
     @LoggingAspectConfig(resultDetails = false)
     public String getEmailTemplate(String tenantKey, String templatePath, String langKey) {
         return getTemplateOverrideable(tenantKey, templatePath, langKey).orElseThrow(() -> new IllegalArgumentException("Email template was not found"));
     }
 
+    @LoggingAspectConfig(resultDetails = false)
+    public String getEmailTemplateByKey(TenantKey tenantKey, String templateKey, String locale) {
+        Optional<EmailTemplateSpec> emailTemplateSpec = emailSpecService.getEmailTemplateSpec(tenantKey.getValue(), templateKey);
+        if (emailTemplateSpec.isPresent()) {
+            String templatePath = emailTemplateSpec.get().getTemplatePath();
+            Optional<String> emailTemplateOptional =  getTemplateOverrideable(templatePath);
+
+            return emailTemplateOptional.orElseGet(() -> getEmailTemplate(tenantKey.getValue(), templateKey, locale));
+        }
+        return getEmailTemplate(tenantKey.getValue(), templateKey, locale);
```

Comment:
emailSpecService.getEmailTemplateSpec(tenantKey.getValue(), templateKey)
  .map(EmailTemplateSpec::getTemplatePath)
  .map(this::getTemplateOverrideable)
  .orElseGet(() -> getEmailTemplate(tenantKey.getValue(), templateKey, locale));